controllers/v1/surveysController.js

```diff
@@ -0,0 +1,709 @@
+     * @apiParam {File} survey  Mandatory survey file of type CSV.
+     * @apiUse successBody
+     * @apiUse errorBody
+     */
+
+    /**
+    * Upload bulk surveys via csv.
+    * @method
+    * @name bulkCreate
+    * @param {Object} req -request Data.
+    * @param {CSV} req.files.survey -Surveys csv data . 
+    * @returns {CSV} - Same uploaded csv with extra field status indicating the particular
+    * column is uploaded or not. 
+    */
+
+   async bulkCreate(req) {
+    return new Promise(async (resolve, reject) => {
+        try {
+            
+            if (!req.files || !req.files.survey) {
+                let responseMessage = httpStatusCode.bad_request.message;
+                return resolve({ 
+                    status: httpStatusCode.bad_request.status, 
+                    message: responseMessage 
+                });
+            }
+
+            const fileName = `Survey-Upload-Result`;
+            let fileStream = new FileStream(fileName);
+            let input = fileStream.initStream();
+
+            (async function () {
+                await fileStream.getProcessorPromise();
+                return resolve({
+                    isResponseAStream: true,
+                    fileNameWithPath: fileStream.fileNameWithPath()
+                });
+            })();
+
+            let surveyData = 
+            await csv().fromString(req.files.survey.data.toString());
+
+            let users = [];
+            let usersKeycloakIdMap = {};
+            let solutionExternalIds = [];
+
+            surveyData.forEach(eachSurveyData => {
+                if (!eachSurveyData["keycloak-userId"] && eachSurveyData.user && !users.includes(eachSurveyData.user)) {
+                    users.push(eachSurveyData.user);
+                } else if (eachSurveyData["keycloak-userId"] && eachSurveyData["keycloak-userId"] != "") {
+                    usersKeycloakIdMap[eachSurveyData["keycloak-userId"]] = true;
+                }
+                solutionExternalIds.push(eachSurveyData.solutionExternalId);
+            })
+           
+            let userIdByExternalId;
+
+            if (users.length > 0) {
+                userIdByExternalId = await assessorsHelper.getInternalUserIdByExternalId(req.rspObj.userToken, users);
+                if(Object.keys(userIdByExternalId).length > 0) {
+                    Object.values(userIdByExternalId).forEach(userDetails => {
+                        usersKeycloakIdMap[userDetails.userId] = true;
+                    })
+                }
+            }
+
+            if(Object.keys(usersKeycloakIdMap).length > 0) {
+                
+                let userOrganisationDetails = await observationsHelper.getUserOrganisationDetails(
+                    Object.keys(usersKeycloakIdMap), 
+                    req.rspObj.userToken
+                );
+
+                usersKeycloakIdMap = userOrganisationDetails.data;
+            }
+            
+            let solutionQuery = {
```

Comment:
@deepakr123 - You should check if they are mapping to solutions which are isReusable false only

Reply:
resolved